templates/common/.github/workflows/bicep/azure-dev.yml

```diff
@@ -24,9 +24,11 @@ jobs:
       AZURE_TENANT_ID: ${{ vars.AZURE_TENANT_ID }}
       AZURE_SUBSCRIPTION_ID: ${{ vars.AZURE_SUBSCRIPTION_ID }}
       AZURE_CREDENTIALS: ${{ secrets.AZURE_CREDENTIALS }}
+      AZURE_ENV_NAME: ${{ vars.AZURE_ENV_NAME }}
+      AZURE_LOCATION: ${{ vars.AZURE_LOCATION }}
```

Comment:
Let's keep the variables at the step level. While they both work the same, the step-level variables make it clearer what is required to run `azd provision` or `azd deploy`.

Reply:
It makes the workflow longer, thus harder to follow along. I believe that moving these at the job-level is best, as it says all of these are required for the job to work.

You have to ask yourself, "are we trying to teach people what the args are for `azd provision` and `azd deploy` commands or are we trying to give them a clean, functional solution with a minimal cognitive load?" Again, I think this template is too complex and this helps simplify it some.

Reply:
I would keep any variable on the top, but secrets at the step level.  
secrets should be limited as much as possible to a particular audience as a good practice.